agent/xds/clusters.go

```diff
@@ -658,6 +658,26 @@ func (s *ResourceGenerator) clustersFromSnapshotIngressGateway(cfgSnap *proxycfg
 			}
 
 			for _, c := range upstreamClusters {
+				// Adjust the limit for upstream service
+				// Lookup listener and service config details from ingress gateway
+				// definition.
+				var svc *structs.IngressService
+				if lCfg, ok := cfgSnap.IngressGateway.Listeners[listenerKey]; ok {
+					svc = findIngressServiceMatchingUpstream(lCfg, u)
+				}
+
+				if svc != nil && svc.MaxConnections > 0 {
+					if c.CircuitBreakers.Thresholds == nil {
+						c.CircuitBreakers.Thresholds = []*envoy_cluster_v3.CircuitBreakers_Thresholds{
+							{
+								MaxConnections: makeUint32Value(int(svc.MaxConnections)),
+							},
+						}
+					} else {
```

Comment:
Actually the else branch will never happen because 

```
* The Proxy.Upstreams configuration is invalid for a ingress-gateway
```

Leave it here for safety check.